tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -0,0 +1,756 @@
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the nic: %s", nicName))
+		return
+	}
+	var response bool = false
+	var responseStatus int = 200
+	var r autorest.Response
+	for i := 1; i <= 300; i++ {
+		time.Sleep(1 * time.Second)
+		r, err = future.Result(az.nicClient)
+		response = r.IsHTTPStatus(responseStatus)
+		if response {
+			return nil
+		}
+	}
+	return err
+}
+
+// destroyNSG deletes the security groups by taking it's name as argument.
+func (az *AzureProvider) destroyNSG(ctx context.Context, nsgName string) (err error) {
+	future, err := az.nsgClient.Delete(ctx, az.resourceGroupName, nsgName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the sec group: %s", nsgName))
+		return
+	}
+	var response bool = false
+	var responseStatus int = 200
+	var r autorest.Response
+	for i := 1; i <= 300; i++ {
+		time.Sleep(1 * time.Second)
+		r, err = future.Result(az.nsgClient)
+		response = r.IsHTTPStatus(responseStatus)
+		if response {
+			return nil
+		}
+	}
+	return err
+}
+
+// destroyDisk deletes the disk attached with the instance by taking its name
+// as an argument.
+func (az *AzureProvider) destroyDisk(ctx context.Context, vmInfo compute.VirtualMachine) (err error) {
+	vmStorageProfile := *(vmInfo.VirtualMachineProperties.StorageProfile)
+	vmOSdiskProperties := *(vmStorageProfile.OsDisk)
+	diskName := *(vmOSdiskProperties.Name)
+	az.diskClient.PollingDuration = time.Minute * 10
+	future, err := az.diskClient.Delete(ctx, az.resourceGroupName, diskName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the root disk: %s", diskName))
+		return
+	}
+	var response bool = false
+	var responseStatus int = 200
+	var r autorest.Response
+	for i := 1; i <= 300; i++ {
+		time.Sleep(1 * time.Second)
+		r, err = future.Result(az.diskClient)
+		response = r.IsHTTPStatus(responseStatus)
+		if response {
+			return nil
+		}
+	}
+	return err
+}
+
+// DestroyWindowsVMs destroys all the resources created by the CreateWindows method.
+func (az *AzureProvider) DestroyWindowsVMs() error {
+	// Read from `windows-node-installer.json` file
+	log.Info(fmt.Sprintf("processing file '%s'", az.resourceTrackerDir))
+	ctx := context.Background()
+	resourceTrackerFilePath, err := resource.MakeFilePath(az.resourceTrackerDir)
+	if err != nil {
+		return err
+	}
+	installerInfo, err := resource.ReadInstallerInfo(resourceTrackerFilePath)
+	if err != nil {
+		return fmt.Errorf("unable to get saved info from json file")
+	}
+	var terminatedInstances, deletedSg []string
+
+	for _, vmName := range installerInfo.InstanceIDs {
+
+		_, nicName := az.getNICname(ctx, vmName)
+		_, ipName := az.getIPname(ctx, vmName)
+
+		vmInfo, _ := az.vmClient.Get(ctx, az.resourceGroupName, vmName, compute.InstanceView)
+
+		log.Info(fmt.Sprintf("deleting the resource associated with the instance: %s", vmName))
+
+		err = az.destroyInstance(ctx, vmName)
+		if err == nil {
+			log.Info(fmt.Sprintf("deleted the instance '%s'", vmName))
```

Comment:
Can you destroy the other resources if there was an error destroying the instance?

Reply:
Nope, I can't delete the IP unless the NIC is deleted, same way with instance. I will take care of this in the next commit.